azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureProxy.java

```diff
@@ -141,84 +139,83 @@ protected Object handleAsyncHttpResponse(final HttpRequest httpRequest, Single<H
-                                pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                                if (pollStrategy == null) {
-                                    result = createProvisioningStateOrCompletedPollStrategy(originalHttpRequest, originalHttpResponse, methodParser, delayInMilliseconds);
-                                }
-                            } else if (httpStatusCode == 202) {
-                                pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                                if (pollStrategy == null) {
-                                    pollStrategy = LocationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, delayInMilliseconds);
-                                }
-                            }
-                        }
-                        else {
-                            if (httpStatusCode == 202) {
-                                pollStrategy = AzureAsyncOperationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, expectsResourceResponse, serializer, delayInMilliseconds);
-                                if (pollStrategy == null) {
-                                    pollStrategy = LocationPollStrategy.tryToCreate(AzureProxy.this, originalHttpRequest, originalHttpResponse, delayInMilliseconds);
-                                    if (pollStrategy == null) {
-                                        throw new CloudException("Response does not contain an Azure-AsyncOperation or Location header.", originalHttpResponse);
+                        final int[] longRunningOperationStatusCodes = new int[] {200, 201, 202};
+                        return ensureExpectedStatus(originalHttpResponse, methodParser, longRunningOperationStatusCodes)
+                                .flatMap(new Func1<HttpResponse, Single<? extends PollStrategy>>() {
+                                    @Override
+                                    public Single<? extends PollStrategy> call(HttpResponse response) {
+                                        Single<PollStrategy> result = null;
+
+                                        final Long parsedDelayInMilliseconds = PollStrategy.delayInMillisecondsFrom(originalHttpResponse);
+                                        final long delayInMilliseconds = parsedDelayInMilliseconds != null ? parsedDelayInMilliseconds : AzureProxy.defaultDelayInMilliseconds();
+
+                                        final String originalHttpRequestMethod = originalHttpRequest.httpMethod();
+
+                                        PollStrategy pollStrategy = null;
+                                        if (httpStatusCode == 200) {
```

Comment:
It's probably not desirable to do the work to change it now, but I think a more repetitive style of conditions in if statements could be easier to read. Each if statement you go deeper into requires you to accumulate in your mind what assumptions can be made inside the block, and inside that child block, etc.